src/Tools/ByteArrayComparer.cs

```diff
@@ -14,7 +13,10 @@ public bool Equals(byte[] left, byte[] right)
 
         public int GetHashCode(byte[] key)
         {
-            return new BigInteger(key).GetHashCode();
+            uint hash = 0;
+            for (int i = key.Length - 1; i >= 0; i--)
+                hash = (key[i] + hash + 1) * 2654435761;
+            return (int)hash;
```

Comment:
I used to use this tool but now words are referenced as their first locations in the buffer, cause it uses less memory and hashes faster.

I decided to keep this tool in case somebody needs it. This seems to be a faster implementation, I'll take your word for it that it works.